drivers/periph_common/init.c

```diff
@@ -60,6 +63,52 @@
 #ifdef MODULE_PERIPH_INIT_SDMMC
 #include "sdmmc/sdmmc.h"
 #endif
+
+#if IS_USED(MODULE_PERIPH_INIT_I2C)
+#include "periph/gpio.h"
+#include "busy_wait.h"
+
+static void _i2c_sync_devices(i2c_t dev)
+{
+    /* Recover possible bus hangup by clocking peripheral
+     * i2c device state machines into idle.
+     * Hangup can occur if a transaction was interrupted
+     * by a reset of the MCU.
+     * Badly designed hardware can also cause a hangup
+     * due to glitches on the bus during operation which
+     * we do not aim to mitigate here.
+     */
+    gpio_init(i2c_config[dev].sda_pin, GPIO_IN);
+    gpio_init(i2c_config[dev].scl_pin, GPIO_OUT);
+    gpio_set(i2c_config[dev].scl_pin);
+    if (IS_ACTIVE(ENABLE_DEBUG)) {
+        if (!gpio_read(i2c_config[dev].sda_pin)) {
+            DEBUG("periph_init: i2c bus hangup detected for bus #%u,"
+                " recovering...\n", dev);
+        }
+        else {
+            DEBUG("periph_init: i2c bus #%u is ok\n", dev);
```

Comment:
If the bus is "OK" Aren't we suppose to skip the next clock ?

Reply:
The clock is intended to be skipped for the gpio_read returning 1 in the while loop below. This section is only for purposes of debug output.

Reply:
Sorry I didn't get my morning coffee yet...
I mean "the next BLOCK".
If the bus is OK, couldn't we skip entirely the whole loop that comes after.